pkg/kubectl/cmd/run.go

```diff
@@ -220,18 +220,13 @@ func Run(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobr
 		case api.RestartPolicyOnFailure:
 			if contains(resourcesList, batchv1.SchemeGroupVersion.WithResource("jobs")) {
 				generatorName = "job/v1"
-			} else if contains(resourcesList, v1beta1.SchemeGroupVersion.WithResource("jobs")) {
```

Comment:
We'll need to update generator doc as well http://kubernetes.io/docs/user-guide/kubectl-conventions/#generators (in a separate PR and in the next release, of course)